testing/go/functions_test.go

```diff
@@ -1264,5 +1239,191 @@ func TestDateAndTimeFunction(t *testing.T) {
+					Expected: []sql.Row{{float64(10000000)}},
+				},
+				{
+					Query:    `SELECT EXTRACT(MILLENNIUM FROM INTERVAL '2001 years');`,
+					Expected: []sql.Row{{float64(2)}},
+				},
+				{
+					Query:    `select extract(millenniums from interval '3000 years 65 minutes 10 seconds');`,
+					Expected: []sql.Row{{float64(3)}},
+				},
+				{
+					Query:    `select extract(millisecond from interval '10 months 65 minutes 10 seconds');`,
+					Expected: []sql.Row{{float64(10000.000)}},
+				},
+				{
+					Query:    `select extract(minutes from interval '10 months 65 minutes 10 seconds');`,
+					Expected: []sql.Row{{float64(5)}},
+				},
+				{
+					Query:    `SELECT EXTRACT(MONTH FROM INTERVAL '2 years 3 months');`,
+					Expected: []sql.Row{{float64(3)}},
+				},
+				{
+					Query:    `SELECT EXTRACT(MONTH FROM INTERVAL '2 years 13 months');`,
+					Expected: []sql.Row{{float64(1)}},
+				},
+				{
+					Query:    `select extract(months from interval '20 months 65 minutes 10 seconds');`,
+					Expected: []sql.Row{{float64(8)}},
+				},
+				{
+					Query:    `select extract(quarter from interval '20 months 65 minutes 10 seconds');`,
+					Expected: []sql.Row{{float64(3)}},
+				},
+				{
+					Query:    `select extract(seconds from interval '65 minutes 10 seconds 5 millisecond');`,
+					Expected: []sql.Row{{float64(10.005000)}},
+				},
+				{
+					Query:    `select extract(years from interval '20 months 65 minutes 10 seconds');`,
+					Expected: []sql.Row{{float64(1)}},
+				},
+			},
+		},
+		{
+			Name:        "age",
+			SetUpScript: []string{},
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:    `SELECT age(timestamp '2001-04-10', timestamp '1957-06-13');`,
+					Expected: []sql.Row{{"43 years 9 mons 27 days"}},
+				},
+				{
+					Query:    `SELECT age(timestamp '1957-06-13', timestamp '2001-04-10');`,
+					Expected: []sql.Row{{"-43 years -9 mons -27 days"}},
+				},
+				{
+					Query:    `SELECT age(timestamp '2001-06-13', timestamp '2001-04-10');`,
+					Expected: []sql.Row{{"2 mons 3 days"}},
+				},
+				{
+					Query:    `SELECT age(timestamp '2001-04-10', timestamp '2001-06-13');`,
+					Expected: []sql.Row{{"-2 mons -3 days"}},
+				},
+				{
+					Query:    `SELECT age(timestamp '2001-04-10 12:23:33', timestamp '1957-06-13 13:23:34.4');`,
+					Expected: []sql.Row{{"43 years 9 mons 26 days 22:59:58.6"}},
+				},
+				{
+					Query:    `SELECT age(timestamp '1957-06-13 13:23:34.4', timestamp '2001-04-10 12:23:33');`,
+					Expected: []sql.Row{{"-43 years -9 mons -26 days -22:59:58.6"}},
+				},
+				{
+					Skip:     true, // TODO: current_date should return timestamp, not text
+					Query:    `SELECT age(current_date);`,
+					Expected: []sql.Row{{"00:00:00"}},
+				},
+				{
+					Query:    `SELECT age(current_date::timestamp);`,
+					Expected: []sql.Row{{"00:00:00"}},
+				},
+			},
+		},
+		{
+			Name: "Interval type functions",
```

Comment:
We should move these to the `operators_test.go` file. Also, each operator should at least have a `true` and `false` result, and those that do not return boolean values should also have at least two distinct enough tests.